drydock/plugin.py

```diff
@@ -1,11 +1,179 @@
+        pre_init_tasks.append((service, tutor_env.read_template_file(*init_path)))
+
+    init_tasks = pre_init_tasks + init_tasks
+
+    for service, init_path in list(tutor_hooks.Filters.COMMANDS_INIT.iterate()):
+        init_tasks.append((service, tutor_env.read_template_file(*init_path)))
+
+    response = []
+    for i, (service, command) in enumerate(init_tasks):
+        for template in _load_jobs(tutor_conf):
+            if template['metadata']['name'] != service + '-job':
+                continue
+
+            render_command = tutor_env.render_str(tutor_conf, command)
+
+            template['metadata']['name'] = 'drydock-' + template['metadata']['name'] + '-' + str(i)
+            template['metadata']['labels'] = {
+                'drydock.io/component': 'job',
+                'drydock.io/target-service': template['metadata']['name'],
+                'drydock.io/runner-service': template['metadata']['name']
+            }
+            template['metadata']['annotations'] = {
+                'argocd.argoproj.io/sync-wave': INIT_JOBS_SYNC_WAVE + i,
+                'argocd.argoproj.io/hook': 'Sync',
+                'argocd.argoproj.io/hook-delete-policy': 'HookSucceeded'
+            }
+
+            shell_command = ["sh", "-e", "-c"]
+            if template["spec"]["template"]["spec"]["containers"][0].get("command") == []:
+                # In some cases, we need to bypass the container entrypoint.
+                # Unfortunately, AFAIK, there is no way to do so in K8s manifests. So we mark
+                # some jobs with "command: []". For these jobs, the entrypoint becomes "sh -e -c".
+                # We do not do this for every job, because some (most) entrypoints are actually useful.
+                template["spec"]["template"]["spec"]["containers"][0]["command"] = shell_command
+                container_args = [render_command]
+            else:
+                container_args = shell_command + [render_command]
+
+            template["spec"]["template"]["spec"]["containers"][0]["args"] = container_args
+            template["spec"]["backoffLimit"] = 1
+            template["spec"]["ttlSecondsAfterFinished"] = 3600
+
+
+            yield serialize.dumps(template)
+
+    return response
+
+
+def get_upgrade_list():
+    """
+    Return a list of upgrade target versions based on the MIGRATE_FROM setting.
+    """
+    context = click.get_current_context().obj
+    tutor_conf = tutor_config.load(context.root)
+
+    upgrade_list = []
+    if not 'MIGRATE_FROM' in tutor_conf:
+        return upgrade_list
+    migrate_from = tutor_conf["MIGRATE_FROM"]
+    migrate_to = config['defaults']['VERSION'].split('.', maxsplit=1)[0]
+    for name, version in VERSION_LIST:
+        if migrate_from.lower() == name.lower():
+            migrate_from = version
+        if migrate_from <= version <= migrate_to:
+            upgrade_list.append(name.lower())
+    return upgrade_list
```

Comment:
Maybe this could be simpler if we only used numerical versions (only the mayor)?